### PR TITLE
fix(query): fix for schema mismatch error in binaryOp with fixed scalar value

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?

**Current behavior :**
BinaryJoin queries with RangeVector and ScalarVector combination are failing with following error
> filodb.core.memstore.SchemaMismatch: Multiple schemas found, please filter. Expected schema..

**New behavior :**
ResultSchema comparison is made less restrictive to exclude `fixedVectorLen` and `coldIDs`

### CODE REVIEW

File: query/src/main/scala/filodb/query/LogicalPlan.scala [hunk 1]
```diff
@@ -274,9 +274,9 @@ final case class ScalarVaryingDoublePlan(vectors: PeriodicSeriesPlan,
   */
 final case class ScalarTimeBasedPlan(function: ScalarFunctionId, rangeParams: RangeParams) extends ScalarPlan {
   override def isRoutable: Boolean = false
-  override def startMs: Long = rangeParams.start
-  override def stepMs: Long = rangeParams.step
-  override def endMs: Long = rangeParams.end
+  override def startMs: Long = rangeParams.start * 1000
+  override def stepMs: Long = rangeParams.step * 1000
+  override def endMs: Long = rangeParams.end * 1000
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/LogicalPlan.scala]: Thanks for the fix! Can we rename rangeParams fields to `startSecs`, `endSecs` etc ?

File: query/src/main/scala/filodb/query/exec/ExecPlan.scala [hunk 2]
```diff
@@ -406,7 +406,7 @@ abstract class NonLeafExecPlan extends ExecPlan {
       case QueryResult(_, schema, _) if rs == ResultSchema.empty =>
         schema     /// First schema, take as is
       case QueryResult(_, schema, _) =>
-        if (rs != schema) throw SchemaMismatch(rs.toString, schema.toString)
+        if (rs.hasSameColumnsAs(schema)) throw SchemaMismatch(rs.toString, schema.toString)
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/ExecPlan.scala]: This is tricky and needs more thought. 

Does this mean we can join counters and gauges ?

Comment (vishramachandran) [reply query/src/main/scala/filodb/query/exec/ExecPlan.scala]: Can we make this apply this fix only to joins ?
Doing it here, will apply for dist-concat as well...

Comment (sherali42) [reply query/src/main/scala/filodb/query/exec/ExecPlan.scala]: ack, fixed

File: query/src/main/scala/filodb/query/LogicalPlan.scala (hunk 1 above)
Comment (sherali42) [reply query/src/main/scala/filodb/query/LogicalPlan.scala]: ack, fixed it

File: core/src/main/scala/filodb.core/query/ResultTypes.scala [hunk 3]
```diff
@@ -52,9 +52,9 @@ final case class ResultSchema(columns: Seq[ColumnInfo], numRowKeyColumns: Int,
                               columns(1).colType == HistogramColumn && columns(2).colType == DoubleColumn
 
   def hasSameColumnsAs(other: ResultSchema): Boolean = {
-    // exclude fixedVectorLen
+    // exclude fixedVectorLen & coldIDs
```
[the remark this answers is not in the document]
Comment (sherali42) [reply core/src/main/scala/filodb.core/query/ResultTypes.scala]: fixed

File: query/src/main/scala/filodb/query/exec/ExecPlan.scala [hunk 4]
```diff
@@ -378,8 +378,7 @@ abstract class NonLeafExecPlan extends ExecPlan {
     // an empty schema.  Validate that the other schemas are the same.  Skip over empty schemas.
     var sch = ResultSchema.empty
     val processedTasks = childTasks.collect {
-      case (res @ QueryResult(_, schema, _), i) if schema != ResultSchema.empty =>
-        sch = reduceSchemas(sch, res)
+      case (res @ QueryResult(_, schema, _), i) if schema != ResultSchema.empty => sch = reduceSchemas(sch, res)
```
[the remark this answers is not in the document]
Comment (sherali42) [reply query/src/main/scala/filodb/query/exec/ExecPlan.scala]: reverted the change

**Pull request merged**